urllib3/response.py

```diff
@@ -319,6 +319,8 @@ def from_httplib(ResponseCls, r, **response_kw):
         with ``original_response=r``.
         """
         headers = r.msg
+        validate_headers(headers)
```

Comment:
If we feel strongly it should return a boolean rather than raise, then I'd follow the `is_foo()` pattern we use elsewhere too.